xcdat/dataset.py

```diff
@@ -779,20 +633,22 @@ def _get_cftime_coords(
     np.ndarray
         An array of `cftime` coordinates.
     """
-    # Starting from the reference date, create an array of `datetime` objects
-    # by adding each offset (a numerically encoded value) to the reference date.
-    # The `parse.parse` default is set to datetime(2000, 1, 1), with each
-    # component being a placeholder if the value does not exist. For example, 1
-    # and 1 are placeholders for month and day if those values don't exist.
+    # Create an array of `datetime` objects by adding the number offset to the
+    # reference date. The `parse.parse` default is set to datetime(2000, 1, 1),
+    # with each component being a placeholder if the value does not exist.
     ref_datetime: datetime = parser.parse(ref_date, default=datetime(2000, 1, 1))
-    offsets = np.array(
+
+    # Convert offsets to `np.float64` to avoid "TypeError: unsupported type for
+    # timedelta days component: numpy.int64".
+    offsets = offsets.astype("float64")
+    times = np.array(
         [ref_datetime + rd.relativedelta(**{units: offset}) for offset in offsets],
         dtype="object",
     )
 
     # Convert the array of `datetime` objects into `cftime` objects based on
     # the calendar type.
     date_type = get_date_type(calendar)
```

Comment:
if units in ["months", "years"]:
        # Convert offsets to `np.float64` to avoid "TypeError: unsupported type for
        # timedelta days component: numpy.int64".
        offsets = offsets.astype("float64")
        times = np.array(
            [ref_datetime + rd.relativedelta(**{units: offset}) for offset in offsets],
            dtype="object",
        )

        # Convert the array of `datetime` objects into `cftime` objects based on
        # the calendar type.
        date_type = get_date_type(calendar)
        coords = convert_times(times, date_type=date_type)
    else:
        xrunits = units + " since " + ref_date
        coords = xr.coding.times.decode_cf_datetime(offsets, xrunits, calendar=calendar, use_cftime=True)
```suggestion
    # Convert offsets to `np.float64` to avoid "TypeError: unsupported type for
    # timedelta days component: numpy.int64".
    offsets = offsets.astype("float64")
    times = np.array(
        [ref_datetime + rd.relativedelta(**{units: offset}) for offset in offsets],
        dtype="object",
    )

    # Convert the array of `datetime` objects into `cftime` objects based on
    # the calendar type.
    date_type = get_date_type(calendar)
```